nan_typedarray_contents.h

```diff
@@ -57,19 +57,19 @@ class TypedArrayContents {
 
 #endif
 
-#if defined(_MSC_VER) || defined(__GNUC__)
+#if defined(_MSC_VER) && _MSC_VER >= 1900 || __cplusplus >= 201103L
```

Comment:
However, I see I could break this up into two commits to make things more logical.